### PR TITLE
Rebuild with introspection for gst-plugins-base

### LINKED ISSUE #21: Namespace GstAudio not available
Issue:

I can't load the `GstAudio` plugin:

```pytb
$ python -c "import gi; gi.require_version('GstAudio', '1.0')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/duncan.macleod/opt/miniconda3/envs/gsttest/lib/python3.6/site-packages/gi/__init__.py", line 130, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace GstAudio not available
```

Is this supposed to be provided by the `gst-plugins-base` package?

If I compare the conda package to the (hopefully equivalent) `gstreamer1-plugings-base` RPM, I see that the conda packge doesn't provide `lib/girepository-1.0/GstAudio-1.0.typelib`. Is that the issue?

I know nothing about the gstreamer ecosystem, so I'm sorry if I'm talking nonsense here.

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /home/duncan.macleod/opt/miniconda3/envs/gsttest:
#
# Name                    Version                   Build  Channel
ca-certificates           2018.11.29           ha4d7672_0    conda-forge
cairo                     1.14.12           h80bd089_1005    conda-forge
fontconfig                2.13.1            h2176d3f_1000    conda-forge
freetype                  2.9.1             h94bbf69_1005    conda-forge
gettext                   0.19.8.1          h9745a5d_1001    conda-forge
glib                      2.56.2            had28632_1001    conda-forge
gobject-introspection     1.56.1          py36h9e29830_1001    conda-forge
gst-plugins-base          1.12.5            h3865690_1000    conda-forge
gstreamer                 1.12.5            h0cc0488_1000    conda-forge
icu                       58.2              hf484d3e_1000    conda-forge
libffi                    3.2.1             hf484d3e_1005    conda-forge
libgcc-ng                 7.3.0                hdf63c60_0    conda-forge
libiconv                  1.15              h14c3975_1004    conda-forge
libpng                    1.6.36            h84994c4_1000    conda-forge
libstdcxx-ng              7.3.0                hdf63c60_0    conda-forge
libuuid                   2.32.1            h14c3975_1000    conda-forge
libxcb                    1.13              h14c3975_1002    conda-forge
libxml2                   2.9.8             h143f9aa_1005    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
openssl                   1.0.2p            h14c3975_1002    conda-forge
pcre                      8.41              hf484d3e_1003    conda-forge
pixman                    0.34.0            h14c3975_1003    conda-forge
pthread-stubs             0.4               h14c3975_1001    conda-forge
pycairo                   1.18.0          py36h1b9232e_1000    conda-forge
pygobject                 3.28.3          py36h89f6ae1_1001    conda-forge
python                    3.6.7             hd21baee_1001    conda-forge
readline                  7.0               hf8c457e_1001    conda-forge
sqlite                    3.26.0            h67949de_1000    conda-forge
tk                        8.6.9             h84994c4_1000    conda-forge
xorg-kbproto              1.0.7             h14c3975_1002    conda-forge
xorg-libice               1.0.9             h14c3975_1004    conda-forge
xorg-libsm                1.2.3             h4937e3b_1000    conda-forge
xorg-libx11               1.6.6             h14c3975_1000    conda-forge
xorg-libxau               1.0.8             h14c3975_1006    conda-forge
xorg-libxdmcp             1.1.2             h14c3975_1007    conda-forge
xorg-libxext              1.3.3             h14c3975_1004    conda-forge
xorg-libxrender           0.9.10            h14c3975_1002    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h14c3975_1002    conda-forge
xorg-xproto               7.0.31            h14c3975_1007    conda-forge
xz                        5.2.4             h14c3975_1001    conda-forge
zlib                      1.2.11            h14c3975_1004    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : gsttest
    active env location : /home/duncan.macleod/opt/miniconda3/envs/gsttest
            shell level : 1
       user config file : /home/duncan.macleod/.condarc
 populated config files : /home/duncan.macleod/.condarc
          conda version : 4.5.12
    conda-build version : 3.17.6+12.g4fc12f89
         python version : 3.7.1.final.0
       base environment : /home/duncan.macleod/opt/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /home/duncan.macleod/opt/miniconda3/pkgs
                          /home/duncan.macleod/.conda/pkgs
       envs directories : /home/duncan.macleod/opt/miniconda3/envs
                          /home/duncan.macleod/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.12 requests/2.21.0 CPython/3.7.1 Linux/3.10.0-957.1.3.el7.x86_64 scientific/7.6 glibc/2.17
                UID:GID : 5308:5308
             netrc file : /home/duncan.macleod/.netrc
           offline mode : False

```
</details>

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR rebuilds this feedstock with the added `--enable-introspection` flag for `gst-plugins-base`.

Fixes #21.

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
@conda-forge/gstreamer, this build is failing with:

```
/home/conda/feedstock_root/build_artifacts/gstreamer_and_plugins_1548180943066/work/plugins_base/gst-libs/gst/gl/tmp-introspect4lryvyh4/.libs/lt-GstGL-1.0: error while loading shared libraries: libGL.so.1: cannot open shared object file: No such file or directory
```

despite the fact that `mesa-libgl` CDT is being installed. So, I have no idea why this is failing, given that the dependencies are well-specified (AFAIK). The build passes when I run the `.circleci/run_docker_build.sh` script on my own machine. Can someone try and run this on another machine to reproduce the failure? Or just rerun the CI?

**Comment by mingwandroid** [comment]
Did you install the system libGL-providing packages too? You need to. CDTs are only for consumption by the static linker. They are proxies for real system libraries. To ring software you linked against a CDT you need to provide the system libs in the expected system lib location.

**Comment by duncanmmacleod** [comment]
@mingwandroid, can you explain a little more? How would I go about installing system packages for the CI? Is that what `yum_requirements.txt` is for?

Can you explain how this could have worked on my own run of docker but not during CI?

And given that I didn't add the `mesa-libgl-devel` dependency, why has this not errored before?

So many questions; I'm sorry for my ignorance of how this builds, I'm just trying to build a downstream package and know that I need gst-plugins-base build with this new flag, but not much more than that.

**Comment by mingwandroid** [comment]
I am about to take off on a fight.  It is documented though some or ask on gitter

**Comment by duncanmmacleod** [comment]
@conda-forge/gstreamer, I think this is ready, I added `yum_requirements.txt` to address the missing system package, I hope that's the right solution.

**Comment by duncanmmacleod** [comment]
@conda-forge/gstreamer, I think this is now ready for review, thanks.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -77,6 +77,9 @@ outputs:
         - {{ cdt('libxdamage-devel') }}      # [linux]
         - {{ cdt('libxfixes-devel') }}       # [linux]
         - {{ cdt('libxxf86vm-devel') }}      # [linux]
+        - {{ cdt('mesa-dri-drivers') }}      # [linux]
+        - {{ cdt('expat') }}                 # [linux]
```
Comment (ccordoba12) [inline recipe/meta.yaml]: There's a conda-forge package for expat, so why don't we use it instead of the cdt?

Comment (duncanmmacleod) [reply recipe/meta.yaml]: I'm not sure, I don't properly understand what is kosher for CDT and not, I think the libexpat call was coming from a CDT library, but maybe I just totally misread the traceback.

Comment (mingwandroid) [reply recipe/meta.yaml]: This is a dependency needed by the cdt packages. This should be kept.  It didn't imply expat is installed or needed in end user machines.

Comment (ccordoba12) [reply recipe/meta.yaml]: Ok, @duncanmmacleod please add a comment mentioning that above this line. That'll to make it clear that the CDT `expat` is only needed for `mesa-dri-drivers`.

File: recipe/yum_requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+mesa-libGL
```
[the remark this answers is not in the document]
Comment (mingwandroid) [reply recipe/yum_requirements.txt]: Hey Carlos. This is a common misunderstanding with cdts. They are only used by the static linker (ld). If you need to run the built software fort testing or whatnot you need to install the equivalent package via the system package manager. 

If you have no tests (for shame!) then you do not need it. 

For full disclosure you may be able to hack some LD_LIBRARY_PATH stuff but I'd really not recommend that!

[the remark this answers is not in the document]
Comment (ccordoba12) [reply recipe/yum_requirements.txt]: Ok, let's leave it then in case tests are added afterwards.

**Comment by duncanmmacleod** [comment]
@mingwandroid, what's the procedure for duplicating this mod in gst-plugins-base from Anaconda?

**Comment by mingwandroid** [comment]
You can make a pr which syncs the AnacondaRecipes fork with this one.

If x11 is involved (I guess it is without having checked) we need to take care of the differences between cdt and cf-x11 - suggestions welcome - I believe we could each carry a different metapackage (but conda-build needs a small addition to allow transitive run_exports, pinging @msarahan who should be asleep ATM but apparently isn't!)

If x11 isn't involved then a complete sync should be ok/best. We want to minimize differences..

File: recipe/yum_requirements.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (jakirkham) [reply recipe/yum_requirements.txt]: I've added an issue ( https://github.com/conda-forge/conda-forge.github.io/issues/717 ) so we can document this.

**Pull request merged**